client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -835,103 +857,136 @@ def get_template_preset(self):
+        # backward compatibility, since default is True
+        if keep_placeholder is None:
+            keep_placeholder = True
 
         return {
-            "path": solved_path,
+            "path": resolved_path,
             "keep_placeholder": keep_placeholder,
             "create_first_version": create_first_version
         }
 
-    def resolve_template_path(self, path, fill_data) -> str:
+    def resolve_template_path(self, path, fill_data=None) -> str:
         """Resolve the template path.
 
-        By default, this does nothing except returning the path directly.
+        By default, this:
+          - Resolves AYON entity URI to a filesystem path
+          - Returns path directly if it exists on disk.
+          - Resolves template keys through anatomy and environment variables.
 
         This can be overridden in host integrations to perform additional
         resolving over the template. Like, `hou.text.expandString` in Houdini.
+        It's recommended to still call the super().resolve_template_path()
+        to ensure the basic resolving is done across all integrations.
 
         Arguments:
             path (str): The input path.
-            fill_data (dict[str, str]): Data to use for template formatting.
+            fill_data (dict[str, str]): Deprecated. This is computed inside
+                the method using the current environment and project settings.
+                Used to be the data to use for template formatting.
 
         Returns:
             str: The resolved path.
 
         """
-        result = StringTemplate.format_template(path, fill_data)
-        if result.solved:
-            path = result.normalized()
-        return path
+
+        # If the path is an AYON entity URI, then resolve the filepath
+        # through the backend
+        if path.startswith("ayon+entity://") or path.startswith("ayon://"):
+            # This is a special case where the path is an AYON entity URI
+            # We need to resolve it to a filesystem path
+            resolved_path = resolve_entity_uri(path)
+            if not os.path.exists(resolved_path):
+                raise TemplateNotFound(
+                    "Template found in AYON settings for task '{}' with host "
+                    "'{}' does not resolve AYON entity URI '{}' "
+                    "to an existing file on disk: '{}'".format(
+                        self.current_task_name,
+                        self.host_name,
+                        path,
+                        resolved_path,
+                    )
+                )
+            return resolved_path
+
+        # If the path is set and it's found on disk, return it directly
+        if path and os.path.exists(path):
+            return path
+
+        # Otherwise assume a path with template keys, we do a very mundane
+        # check whether `{` or `<` is present in the path.
+        if "{" in path or "<" in path:
```

Comment:
NOTEs:
- Before this change it didn't carsh.
- This condition should check only for `"{"` and revert the logic -> return `path` if there is no `"{"` and fill it if it's not there.

Reply:
Ah you mean - not to raise an error.

The error was previously raised here: https://github.com/ynput/ayon-core/pull/1292/files#diff-96b9bf4cef8323707aa5d129a3c9a93849242fd5b3c7c71ce0d2931f3fa3a46cL902

Reply:
Changed with https://github.com/ynput/ayon-core/pull/1292/commits/7b5ff1394c1684a9a1740ece36e5d343adf853b9

Reply:
We still should rever the condition to first return path, and to check only for `{` (there is no reason to check for `<`).

Reply:
> (there is no reason to check for <).

Aren't those "optional keys" in an anatomy template, or?

Reply:
They're not if `{` is not between them.

- `<{folder[name]}>` -> optional
- `<br>` -> not optional